tools/cdb2sql/cdb2sql.cpp

```diff
@@ -509,6 +515,45 @@ static bool has_delimiter(char *line, int len, char *delimiter, int dlen)
     return true;
 }
 
+char get_delim(){
+    return delimstr[0];
+}
+static const int state_trans[3][3] = {
+                    // character
+//state             // De  WS  Other
+/*Error        - 0*/   {1,  0,  2},
+/*Valid        - 1*/   {1,  1,  2},
+/*Incomplete   - 2*/   {1,  2,  2},
+};
+
+//This function is a state machine that determines
+//if sql is valid
+//
+static bool is_valid_sql(char *sql){
+    int state = 0;
+    char character;
+    while(*sql){
+        char cur_char = *sql;
+        if(cur_char==' ' || cur_char=='\n' ||  cur_char=='\r' || cur_char=='\t'){
+            character = CHAR_WS;
+        }
+        else if(cur_char==get_delim()){
+            character = CHAR_DELIM;
+        }
+        else if(cur_char=='\'' || cur_char == '"'){
```

Comment:
https://www.sqlite.org/lang_keywords.html

There's other types of quotes in the dialect we inherited from SQLite.  They matter less if the goal is to make the output of -gensql be readable back by cdb2sql, but this would be a good place to handle them.

Maybe we can use the sqlite tokenizer here?